plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java

```diff
@@ -568,6 +597,37 @@ protected boolean isLeafMatch(Viewer viewer, Object element) {
         }
     }
 
+    private static class TaskFolderNode {
+        final DBPProject project;
+        final DBTTaskFolder taskFolder;
+
+        TaskFolderNode(DBPProject project, DBTTaskFolder taskFolder) {
+            this.project = project;
+            this.taskFolder = taskFolder;
+        }
+
+        @Override
+        public String toString() {
+            return taskFolder.getName();
+        }
+
+        @Override
+        public int hashCode() {
+            return (project == null ? 0 : project.hashCode()) +
+                (taskFolder == null ? 0 : taskFolder.hashCode());
```

Comment:
Looks like a hand-written implementation which is unsafe because of possible collisions.

Better to implement with the help of IDEA: <kbd>Alt+Insert</kbd> &rArr; <kbd>equals() and hashCode()</kbd>.